src/server/_common.py

```diff
@@ -24,19 +28,51 @@ def _get_db() -> Connection:
 """
 db: Connection = cast(Connection, LocalProxy(_get_db))
 
+@event.listens_for(engine, "before_cursor_execute")
+def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    context._query_start_time = time.time()
+
+
+@event.listens_for(engine, "after_cursor_execute")
+def after_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    # this timing info may be suspect, at least in terms of dbms cpu time...
+    # it is likely that it includes that time as well as any overhead that
+    # comes from throttling or flow control on the streamed data, as well as
+    # any row transform/processing time
+    total_time = time.time() - context._query_start_time
+
+    # Convert to milliseconds
+    total_time *= 1000
+    get_structured_logger('server_api').info("Executed SQL", statement=statement, params=parameters, elapsed_time_ms=total_time)
+
 
 @app.before_request
-def connect_db():
+def before_request_execute():
+    # Set timer for statement
+    g._request_start_time = time.time()
+
+    # Log statement
+    get_structured_logger('server_api').info("Received API request", method=request.method, headers=request.headers, path=request.full_path, args=request.args, user_agent=request.user_agent)
```

Comment:
Applies most of this except `request.form`, because for some reason these are not filled in the `before_request` handler, only `after_request` - i.e
```
# before_request
{"method": "GET", "url": "http://delphi_web_epidata/epidata/api.php?endpoint=fluview_meta", "form_args": {} ... }
# after_request
{"method": "GET", "url": "http://delphi_web_epidata/epidata/api.php?endpoint=fluview_meta", "form_args": {"endpoint": "fluview_meta"} ... }
```

Reply:
`request.form` will be incredibly relevant here when the request comes in with the POST method, which is required when a certain url length is exceeded or at the client's discretion.  if the request parameters come in that way and then are not present in the logs, the logs may be very confusing.

it is also concerning that information from the request is somehow changed during processing...  i dont recall anything in our code that should be overwriting these, so im going to guess that its some strangeness that Flask does.  can you google to see if this is typical Flask behavior?

Reply:
Looked into Flask source code, seems to be intended behavior. Form data is loaded in this method:

https://github.com/pallets/werkzeug/blob/main/src/werkzeug/wrappers/request.py#L251

but it only gets called during the first run of `get_data`, i.e. the actual request execution, so after `before_request`:

https://github.com/pallets/werkzeug/blob/main/src/werkzeug/wrappers/request.py#L374-L410

Reply:
In the current version of the code, `form_args` are still logged in the `@app.after_request` handler; I think that makes sense, as it's properly populated by then.